### DISCUSSION TITLE
Made an App(Narada : MQTT Broker) using moquette for Android

### DISCUSSION THREAD

**Discussion by eswarm** [moquette-io/moquette#684] [Show and tell]
Hello, 

I have made an MQTT broker(using Moquette ofcourse) on Android. Source can be found here, 

https://github.com/eswarm/narada

Play Store link can be found here, 
https://play.google.com/store/apps/details?id=in.eswarm.narada

Thanks for the easy to use library. Will appreciate if the kind folks here can test it or tell me of any bugs.

**Comment by andsel** [comment]
Thanks @eswarm for sharing this!
Moquette just published the version `0.16` if you want to try to embed into your Android application. It redesigned internal concurrency to avoid sharing problems it had in the past, should be more stable.

**Comment by songdehuai** [comment]
Hello, I am a developer from China.I use esp32c3 esp32s3, etc. As the client.Refer to many developers code, found the following questions:

1, use the moquette - broker: 0.16
Esp32 when just electricity will be unable to connect, try again
After the connection fails, fast energizing again, can be normal

2, use the moquette - broker: 0.15
Unable to connect to the above questions no longer appear, but the connection is disconnected for about 10 seconds, after esp32 will repeat connect and disconnect the action

3, in the use of the above problems, in conjunction with the other broker won't appear, such as emqx (https://www.emqx.io/docs/en/v5.0/)

@eswarm I have also tried your code, it's the same phenomenon

I'm just a level application developers, there is no time and energy to explore what happened, if you can, please check this problem.Thank you very much!


哈喽，我是来自中国的开发者。
我使用esp32c3、esp32s3 等作为client。参考很多开发者的代码，发现以下问题：
1，使用moquette-broker:0.16
    esp32在刚通电的时候会无法连接，重试也不行
    连接失败之后，再次快速重新通电，就可以正常了

2，使用moquette-broker:0.15
    上述无法连接的问题不再出现了，但是连接会在大概10秒之内断开，之后esp32会重复连接、断开这个动作

3，在使用以上问题，在使用其他 broker 的时候不会出现，比如 emqx(https://www.emqx.io/docs/en/v5.0/)

@eswarm 我也有试过您的代码，它也是一样的现象

我仅仅是一名应用级开发者，没有时间和精力去探究究竟发生了什么，如果可以的话，请查看此问题。谢谢！

(ESP) https://github.com/songdehuai/ESP_WIFI_TOOLS
(Android) https://github.com/songdehuai/MqttKt

**Reply by songdehuai** [reply]
I finally solved this problem, which was affected by a non-standard mqtt implementation in esp-idf.
https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/protocols/mqtt.html#mqtt-message-retransmission

To solve this problem, only esp_mqtt_client_config_t in esp-idf

`disable_clean_session = true`

Anyway, thank you for your contribution

我最终解决了这个问题，是esp-idf中的一处非标准mqtt实现影响的。
https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/protocols/mqtt.html#mqtt-message-retransmission

解决这个问题只需要在esp-idf中esp_mqtt_client_config_t

`disable_clean_session = true`

无论如何，感谢各位的付出
